### DISCUSSION TITLE
IDs can be confusing.

### DISCUSSION THREAD

**Discussion by excid3** [josefarias/hotwire_combobox#62] [Ideas]
Take this case:

```erb
<%= form_with(model: comment) do |form| %>
    <%= form.combobox :user_id, users_path %>
<% end %>
```

On the server-side response, you need to use:

```erb
<%= async_combobox_options @users, for_id: :comment_user_id %>
```

The `comment_` prefix here is easy to miss andI had to inspect the HTML & turbo stream response to see why the options were not being populated.

Any good ideas how this could be simplified? Rails makes the ID on the generated field automatically, so it's easy to forget that or miss entirely if you're unaware of that feature.

I don't know a great solution for this, but shared context would be nice to make this feel more intuitive.

```erb
<%= async_combobox_options @users, for_id: [Comment.new, :user_id] %>
```

Or maybe the docs just need more examples with these nuances.

**Comment by josefarias** [comment]
Yeah I hear ya. This nuance is mentioned in the docs, but I agree it could be clearer.

> Note that this id is always available in the combobox's data-async-id attribute for easy access, in case you forget it!

Could maybe add more examples in the docs, and/or call out this use case with the form builder explicitly. But I feel like ideally we'd make re-reading the docs unnecessary. Maybe we can implement a custom stream action that'll include the solution in an error message? Very much in line with https://github.com/josefarias/hotwire_combobox/discussions/61

**Reply by excid3** [reply]
What if the request includes the ID in the request as a param? /users?id=comment_user_id

That form helper should be able to determine the ID and then server-side we could avoid specifying it altogether as it could be access from params?

**Reply by josefarias** [reply]
Interesting. Hadn't thought of that. I think that should be possible. Definitely worth a shot.

**Reply by excid3** [reply]
I feel like that approach is used fairly often in Turbo so it would fit well alongside it. I guess it could be a header too, like the Turbo Frame header.

**Reply by josefarias** [reply]
@excid3 the test suite uncovered one case where you might want to specify the id server-side: a single response for multiple combobox streams!

I think we can keep the same API and just use the param as an overridable default. Which I think is what you were thinking but thought this possibility was worth calling out for future reference.

Gotta leave for a bit but will implement this before the next release. This will be a great improvement!

**Reply by excid3** [reply]
Yeah, that should work nicely. Definitely makes sense to make it overridable, but not having to specify it for the default case would be a lot more intuitive on your first try of the gem.

**Comment by josefarias** [comment]
https://github.com/josefarias/hotwire_combobox/pull/64 merged! Thanks for flagging this. It's easy to forget what it's like to use the library for the first time.
